### PR TITLE
Alerting: Simplify notification step

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR introduce the simplification for the notifications step in the alert rule form.
With this PR we finish the work of https://github.com/grafana/alerting-squad/issues/931 

The simplification is protected by the `alertingNotificationsStepMode` feature toggle.

**Why do we need this feature?**

We want to simplify the complexity of the alert rule form flow.

**Who is this feature for?**
All users.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/alerting-squad/issues/931

**Special notes for your reviewer:**

https://github.com/user-attachments/assets/131a6fa2-5652-471f-8599-3b4da1c14779

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/services/featuremgmt/toggles_gen.json [hunk 1]
```diff
@@ -241,6 +241,19 @@
...
       }
     },
+    {
+      "metadata": {
+        "name": "alertingNotificationsStepMode",
+        "resourceVersion": "1730885749857",
+        "creationTimestamp": "2024-11-06T09:35:49Z"
+      },
+      "spec": {
+        "description": "Enables simplified step mode in the nottifications section",
```
Comment (brendamuir) [inline pkg/services/featuremgmt/toggles_gen.json:251]:
```suggestion
        "description": "Enables simplified step mode in the notifications section",
```

**Review by brendamuir — APPROVED** [review]
lgtm text-wise

**Review by tomratcliffe — COMMENTED** [review]
It feels a bit strange having to click the `preview routing` button when you switch to advanced options. Might be a bit fiddly, but would be good for that to be called for the user?

**Comment by soniaAguilarPeiron** [comment]
> It feels a bit strange having to click the `preview routing` button when you switch to advanced options. Might be a bit fiddly, but would be good for that to be called for the user?

This is intentional. There are some performance reasons behind the decision of forcing user to click preview.

File: public/app/features/alerting/unified/components/rule-editor/notificaton-preview/NotificationPreview.tsx [hunk 2]
```diff
@@ -1,11 +1,8 @@
-import { css } from '@emotion/css';
 import { compact } from 'lodash';
 import { lazy, Suspense } from 'react';
 
-import { GrafanaTheme2 } from '@grafana/data';
-import { Button, LoadingPlaceholder, Text, useStyles2 } from '@grafana/ui';
+import { Button, LoadingPlaceholder, Stack, Text } from '@grafana/ui';
 import { alertRuleApi } from 'app/features/alerting/unified/api/alertRuleApi';
-import { Stack } from 'app/plugins/datasource/parca/QueryEditor/Stack';
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rule-editor/notificaton-preview/NotificationPreview.tsx:8]: Not sure why `Stack` was imported from the Parca editor but that seems like an older version that didn't support `justifyContent` etc. so I cleaned that up a bit

**Review by gillesdemey — APPROVED** [review]
Changes LGTM

**Review by Develer — APPROVED** [review]

**Pull request merged**